### DISCUSSION TITLE
No object detection in HASS with coral USB

### DISCUSSION THREAD

**Discussion by Steve973** [blakeblackshear/frigate#4218] [Ask A Question]
Hi, all.  I finally got home assistant set up with the frigate (full access) add-on, and then I was able to get it to find the USB TPU after turning off protected mode.  Though frigate is streaming the cameras perfectly, there is no object detection happening. The cameras are amcrest 4k IP cameras with ai capability, and home assistant is installed in supervised mode.  If I switch to CPU detectors, I get the usual object detections.  With the coral USB, it shows in the log that it found the tpu, and I get no errors, but it's not detecting objects.  How can I begin to troubleshoot this?

**Comment by NickM-27** [comment]
If it is unable to communicate with the coral then frigate will stop. Please paste your entire logs after having frigate running for a bit. 

If you watch the debug camera view with motion and object bounding boxes enabled, what do you see?

**Reply by Steve973** [reply]
After running for over two hours, there is not much in the logs:
![image](https://user-images.githubusercontent.com/16421339/199077459-49679a0d-8531-4ee7-9569-a93e2a513508.png)

In the debug view, I see this:
![image](https://user-images.githubusercontent.com/16421339/199077095-d12d601f-ef2e-499f-9371-2ff462ecd1c1.png)

Note that I have been letting this run for a couple of hours, and I am seeing some, but relatively few, events.

**Reply by NickM-27** [reply]
your inference speed implies that detection is running just fine. I'd suggest looking elsewhere as the coral doesn't seem to be the issue. Again recommend watching the debug view in frigate web UI for the camera to understand what may or may not be going on to your expectations. Also please provide your config file
